src/ng/directive/form.js

```diff
@@ -61,22 +61,28 @@ function nullFormRenameControl(control, name) {
  */
 //asks for $scope to fool the BC controller module
 FormController.$inject = ['$element', '$attrs', '$scope', '$animate', '$interpolate'];
-function FormController(element, attrs, $scope, $animate, $interpolate) {
-  var form = this,
-      controls = [];
+function FormController($element, $attrs, $scope, $animate, $interpolate) {
+  this.$$controls = [];
 
   // init state
-  form.$error = {};
-  form.$$success = {};
-  form.$pending = undefined;
-  form.$name = $interpolate(attrs.name || attrs.ngForm || '')($scope);
-  form.$dirty = false;
-  form.$pristine = true;
-  form.$valid = true;
-  form.$invalid = false;
-  form.$submitted = false;
-  form.$$parentForm = nullFormCtrl;
+  this.$error = {};
+  this.$$success = {};
+  this.$pending = undefined;
+  this.$name = $interpolate($attrs.name || $attrs.ngForm || '')($scope);
+  this.$dirty = false;
+  this.$pristine = true;
+  this.$valid = true;
+  this.$invalid = false;
+  this.$submitted = false;
+  this.$$parentForm = nullFormCtrl;
+
+  this.$$element = $element;
+  this.$$animate = $animate;
```

Comment:
FWIW, We've encountered a similar situation in our codebase when using `.controller` directly since the constructor is what has dependencies injected.  In order to work around this, we've resolved to doing something like this:

```js
angular.module('mod').factory('MyController', function($timeout) {
    return class MyController {
        /* @ngInject */
        constructor() {
            this.waitAMinute().then(() => console.log('Done!'));
        }

        waitAMinute() {
            return $timeout(60000);
        }
    }
});
```

That does require one to inject the controller into the directive and pass the function directly to the controller property rather than using the string syntax, and it also won't work with `ng-controller`.  Just throwing that out there in case it inspires any ideas for future refactors.